OpenRA.Platforms.Default/Sdl2GraphicsContext.cs

```diff
@@ -68,6 +68,27 @@ public IVertexBuffer<Vertex> CreateVertexBuffer(int size)
 			return new VertexBuffer<Vertex>(size);
 		}
 
+		public IIndexBuffer CreateQuadIndexBuffer(int size)
+		{
+			VerifyThreadAffinity();
+
+			if (size % 6 != 0)
+				throw new ArgumentException("Buffer size must be a multiple of 6.");
+
+			var indices = new uint[size];
+			var cornerVertexMap = new uint[] { 0, 1, 2, 2, 3, 0 };
+			for (var i = 0; i < indices.Length; i++)
+				indices[i] = cornerVertexMap[i % 6] + (uint)(4 * (i / 6));
```

Comment:
This still isn't resolved. The indicies have to be populated at the high level call sites. That's `SpriteRenderer` and `TerrainSpriteLayer`. Shuffling this helper around to different places in the low-level renderer doesn't count.

Reply:
I did add a function that allows higher level code to populate it

Reply:
Remove this helper and make those higher-level classes actually define the indicies.

As per my original comment, exposing this helper at all from the low-level rendering code is wrong. The high level code doesn't have to be merely capable of defining the indicies, it is the *only* thing that should be capable of doing so.

Reply:
moved it to Renderer instead